server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/processors/DefineCodeBlock.java

```diff
@@ -14,47 +14,33 @@
  */
 package org.eclipse.lsp.cobol.core.engine.processors;
 
-import org.eclipse.lsp.cobol.common.error.SyntaxError;
-import org.eclipse.lsp.cobol.common.model.tree.CodeBlockDefinitionNode;
-import org.eclipse.lsp.cobol.common.model.tree.Node;
+import org.eclipse.lsp.cobol.common.model.tree.*;
 import org.eclipse.lsp.cobol.common.model.NodeType;
-import org.eclipse.lsp.cobol.common.model.tree.ParagraphsNode;
-import org.eclipse.lsp.cobol.common.model.tree.ProcedureDivisionBodyNode;
 import org.eclipse.lsp.cobol.common.processor.ProcessingContext;
 import org.eclipse.lsp.cobol.common.processor.Processor;
-import org.eclipse.lsp.cobol.core.engine.symbols.SymbolAccumulatorService;
-
-import java.util.List;
-import java.util.Optional;
-
-import static java.util.stream.Collectors.toList;
-import static org.eclipse.lsp.cobol.common.model.tree.Node.hasType;
+import org.eclipse.lsp.cobol.core.engine.symbols.SymbolAccumulator;
 
 /** Processor for ProcedureDivisionBodyNode and ParagraphsNode nodes */
 public class DefineCodeBlock implements Processor<Node> {
-  private final SymbolAccumulatorService symbolAccumulatorService;
+  private final SymbolAccumulator symbolAccumulator;
 
-  public DefineCodeBlock(SymbolAccumulatorService symbolAccumulatorService) {
-    this.symbolAccumulatorService = symbolAccumulatorService;
+  public DefineCodeBlock(SymbolAccumulator symbolAccumulator) {
+    this.symbolAccumulator = symbolAccumulator;
   }
 
   @Override
   public void accept(Node node, ProcessingContext ctx) {
-    if ((node instanceof ProcedureDivisionBodyNode) || (node instanceof ParagraphsNode)) {
-      List<SyntaxError> c =
-          node.getChildren().stream()
-              .filter(hasType(NodeType.PARAGRAPH).or(hasType(NodeType.PROCEDURE_SECTION)))
-              .map(CodeBlockDefinitionNode.class::cast)
-              .map(this::register)
-              .filter(Optional::isPresent)
-              .map(Optional::get)
-              .collect(toList());
-      ctx.getErrors().addAll(c);
+    if (ctx.getCurrentProgramNode() == null) {
+      return;
+    }
+    if ((!(node instanceof ProcedureDivisionBodyNode)) && (!(node instanceof ParagraphsNode))) {
+      return;
+    }
+    for (Node child : node.getChildren()) {
+      if (child.getNodeType() == NodeType.PARAGRAPH
+              || child.getNodeType() == NodeType.PROCEDURE_SECTION) {
+        symbolAccumulator.registerCodeBlock(ctx.getCurrentProgramNode(), (CodeBlockDefinitionNode) child);
```

Comment:
Do you know why we are ignoring the error from `registerCodeBlock`?

Reply:
I don't see how it can produce one.